odoo/tools/date_utils.py

```diff
@@ -163,9 +164,17 @@ def parse_date(value: str, env: Environment) -> date | datetime:
             raise ValueError(f"Invalid term {term!r} in expression date: {value!r}")
 
         # Weekday
-        if (weekday := WEEKDAY_NUMBER.get(term[1:])) is not None:
+        if term[1:] == "week_start":
+            weekday = -1
+        else:
+            weekday = WEEKDAY_NUMBER.get(term[1:])
+
+        if weekday is not None:
+            week_start = int(env["res.lang"]._get_data(code=env.user.lang).week_start) - 1
+            if weekday == -1:
+                weekday = week_start
```

Comment:
```suggestion
        if term[1:] in WEEKDAY_NUMBER or term[1:] == "week_start":
            week_start = int(env["res.lang"]._get_data(code=env.user.lang).week_start) - 1
            weekday = week_start if term[1:] == "week_start" else WEEKDAY_NUMBER[term[1:]]
```
NIT: I feel this is looks simpler, where all variable values are assigned once and we can avoid using `is not None`

Reply:
As long as the JS version looks similar, it is fine for me.